routes/users.js

```diff
@@ -1,9 +1,94 @@
 const express = require('express');
+const { userWidgets, db } = require('../database/schema.js')
 const router = express.Router();
+const mongoose = require('mongoose')
+
 
 /* GET users listing. */
-router.get('/', (request, response, next) => {
-  response.send('respond with a resource');
+router.get('/', ( request, response, next ) => {
+  response.send( 'respond with a resource' );
+});
+
+/* GET widget */
+router.get('/widgets/:widgetId', ( request, response, next ) => {
+  const { widgetId } = request.params
+
+  userWidgets.find( {'widgets._id': widgetId},function(error, widgets){
+      return widgets
+  })
+  .then( data => console.log('finished') )
 });
 
+/* CREATE widgets */
+router.post('/:userId/widgets/create', (request, response, next ) => {
+  const { userId } = request.params
+  const { type, title, size, contents } = request.body
+
+  const newWidget = {
+    type: type,
```

Comment:
We can use implicit object creation:
```js
const widget = { type, title, size, contents }
```

I also prefer a name like `widget` over `newWidget` - the newness should be apparent from the context in which this is executing.  Having said that, I think a better name might be something like `widgetAttributes` or just `attributes`.